siatune/mm/tasks/mmtrainbase.py

```diff
@@ -56,52 +54,22 @@ def train_model(
         """
         pass
 
-    def context_aware_run(self,
-                          searched_cfg,
-                          backend='nccl',
-                          **kwargs) -> None:
-        """Gather and refine the information received by users and Ray.tune to
-        execute the objective task.
-
-        Args:
-            searched_cfg (Config): The searched configs.
-            backend (str):
-                The backend for dist training. Defaults to 'nccl'.
-            kwargs (**kwargs): The kwargs.
-        """
-        # set non blocking mode on the nccl backend
-        # https://github.com/pytorch/pytorch/issues/50820
-        if backend == 'nccl' and os.getenv('NCCL_BLOCKING_WAIT') is None:
-            os.environ['NCCL_BLOCKING_WAIT'] = '0'
-        return super().context_aware_run(searched_cfg, **kwargs)
-
-    def create_trainable(
-        self,
-        backend: str = 'nccl',
-        timeout_s: int = 1800,
-    ) -> ray.tune.trainable:
+    def create_trainable(self) -> TorchTrainer:
         """Get ray trainable task.
 
         Args:
-            backend (str):
-                The backend for dist training. Defaults to 'nccl'.
-            timeout_s (int):
-                Seconds before the torch process group times out.
+            backend (str): The backend for distributed training.
+                Defaults to 'nccl'.
 
         Returns:
-            ray.tune.trainable: The trainable task.
+            TorchTrainer: The trainable task.
         """
-
-        assert backend in ['gloo', 'nccl']
-
-        return DistributedTrainableCreator(
-            partial(
-                self.context_aware_run,
-                backend=backend,
-            ),
-            backend=backend,
-            timeout_s=timeout_s,
-            num_workers=self.num_workers,
-            num_gpus_per_worker=self.num_gpus_per_worker,
-            num_cpus_per_worker=self.num_cpus_per_worker,
-        )
+        return TorchTrainer(
+            self.context_aware_run,
+            scaling_config=ScalingConfig(
+                trainer_resources=dict(
+                    CPU=self.num_cpus_per_worker,
+                    GPU=self.num_gpus_per_worker),
+                num_workers=self.num_workers,
+                use_gpu=torch.cuda.is_available()),
+            torch_config=TorchConfig(backend='gloo'))
```

Comment:
```suggestion
            torch_config=TorchConfig(backend=backend))
```

Reply:
Please refer to [this PR](https://github.com/SIAnalytics/siatune/pull/91)